packages/payment-detection/src/any/any-to-erc20-proxy.ts

```diff
@@ -78,7 +84,7 @@ export class AnyToERC20PaymentDetector extends ERC20FeeProxyPaymentDetectorBase<
     toAddress: string | undefined,
     paymentReference: string,
     requestCurrency: RequestLogicTypes.ICurrency,
-    paymentChain: string,
+    paymentChain: CurrencyTypes.EvmChainName,
```

Comment:
Same in this file. The detection should be agnostic on whether it's from an evm chain or not as long as `ERC20` is supported